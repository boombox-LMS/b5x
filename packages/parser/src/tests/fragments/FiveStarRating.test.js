const {
  exampleFiveStarRatingMarkup,
} = require("../../../dist/componentBuilders/fragments/parsingClasses/FiveStarRating");
const { RawFragmentSchema } = require("@b5x/types");
const { z } = require("zod");

describe("The FiveStarRating fragment builds as expected", () => {
  const fragment = global.buildFragmentFromMarkupString(
    exampleFiveStarRatingMarkup
  );
  const rawFragment = fragment.packageForApi();

  test("The parsed five-star-rating fragment matches the snapshot on file", () => {
    expect(rawFragment).toMatchSnapshot();
  });

  test("The parsed five-star-rating fragment is a valid RawFragment", () => {
    const validator = () => {
      RawFragmentSchema.parse(rawFragment);
    };
    expect(validator).not.toThrowError();
  });

  test("The parsed five-star-rating fragment uses the expected data schema", () => {
    const validator = () => {
      z.object({}).parse(rawFragment.data);
    };
    expect(validator).not.toThrowError();
  });
});
