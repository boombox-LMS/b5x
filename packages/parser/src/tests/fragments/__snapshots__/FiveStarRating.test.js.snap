// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FiveStarRating fragment builds as expected The parsed five-star-rating fragment matches the snapshot on file 1`] = `
Object {
  "childUris": Array [],
  "contentType": "FiveStarRating",
  "contents": "",
  "data": Object {},
  "dependencyUris": Array [],
  "digest": "4e4b92eb709809d174dbc7e147a88888",
  "displayConditions": Array [],
  "documentUri": "mock-document-uri",
  "isRequired": true,
  "isStateful": true,
  "uri": "test-fragment-uri",
}
`;
